src/Chapter2.hs

```diff
@@ -857,7 +877,10 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (x:[]) = [x]
+rewind (x:xs) = (rewind xs) ++ [x]
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Yes, makes sense. I think I may have written this way because I thought of it as a classic linked list reversal problem. I will keep this in mind, thanks.